crates/bevy_core/src/time/fixed_timestep.rs

```diff
@@ -194,3 +195,80 @@ impl System for FixedTimestep {
         self.internal_system.check_change_tick(change_tick);
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use bevy_ecs::prelude::*;
+    use bevy_utils::Instant;
+    use std::ops::{Add, Mul};
+    use std::time::Duration;
+
+    type Count = usize;
+    const LABEL: &str = "test_step";
+
+    #[test]
+    fn test() {
+        let mut world = World::default();
+        let mut time = Time::default();
+        let instance = Instant::now();
+        time.update_with_instant(instance);
+        world.insert_resource(time);
+        world.insert_resource(FixedTimesteps::default());
+        world.insert_resource::<Count>(0);
+        let mut schedule = Schedule::default();
+
+        schedule.add_stage(
+            "update",
+            SystemStage::parallel()
+                .with_run_criteria(FixedTimestep::step(0.5).with_label(LABEL))
+                .with_system(fixed_update),
+        );
+
+        // if time does not progress, the step does not run
+        schedule.run(&mut world);
+        schedule.run(&mut world);
+        assert_eq!(0, *world.get_resource::<Count>().unwrap());
+        assert_eq!(0., get_accumulator_deciseconds(&world));
+
+        // let's progress less than one step
+        set_time(&mut world, instance, 0.4);
+        schedule.run(&mut world);
+        assert_eq!(0, *world.get_resource::<Count>().unwrap());
+        assert_eq!(4., get_accumulator_deciseconds(&world));
+
+        // finish the first step with 0.1s above the step length
+        set_time(&mut world, instance, 0.6);
+        schedule.run(&mut world);
+        assert_eq!(1, *world.get_resource::<Count>().unwrap());
+        assert_eq!(1., get_accumulator_deciseconds(&world));
+
+        // runs multiple times if the delta is multiple step lengths
+        set_time(&mut world, instance, 1.7);
+        schedule.run(&mut world);
+        assert_eq!(3, *world.get_resource::<Count>().unwrap());
+        assert_eq!(2., get_accumulator_deciseconds(&world));
+    }
+
+    fn fixed_update(mut count: ResMut<Count>) {
+        *count += 1;
+    }
+
+    fn set_time(world: &mut World, instance: Instant, seconds: f32) {
```

Comment:
small nit, but this function doesn't set time, it advances time

Reply:
It updates Time with an `Instant`, not a `Duration`. That's why it seems more like setting, not advancing to me.
But I got no problem renaming it.

Reply:
Yeah, I think I agree with `advance_time` here: it's more clear in the context of this test.